### PR TITLE
WIP: Support for `Expect: 100-continue` header

### PR DESCRIPTION
**Pull request by rhuss**
This support is required when uploading data for binary OpenShift builds since this is used by OpenShift for validating a request before accepting the request upload stream.

For this to work, the latest, yet non-released version of OkHttp 3.6.0-SNAPSHOT is required. Support for this feature came in with https://github.com/square/okhttp/commit/a589b8170333476233d48476587a1d4363c90bd0

Since since OkHttp 3.5.0 the WebSocket handling has changed completely, this library needs to be updated for using this new WS scheme. Currently the tests are not running:

* fabric8/mockwebserver needs to be updated to use at least 3.5.0 as well since the tests are using a mixing fabric8/mockwebserver (depending on 3.4.x) and square/okhttp's MockServer (from 3.5.x) which doesn not mix at all.

Also the kubernetes clients needs to removed the depedency on okhttp-ws since the WS support is now directly available within OkHttp directly.

### CODE REVIEW

File: kubernetes-client/pom.xml [hunk 1]
```diff
@@ -170,7 +170,7 @@
     <dependency>
       <groupId>com.squareup.okhttp3</groupId>
       <artifactId>okhttp-ws</artifactId>
-      <version>${okhttp.version}</version>
+      <version>3.4.1</version>
```
Comment (iocanel) [inline kubernetes-client/pom.xml]: why?

File: pom.xml [hunk 2]
```diff
@@ -75,7 +75,8 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <mockwebserver.version>0.0.4</mockwebserver.version>
-    <okhttp.version>3.4.1</okhttp.version>
+    <okhttp.version>3.6.0-SNAPSHOT</okhttp.version>
```
Comment (iocanel) [inline pom.xml]: Ohhh that's why :-)

**Comment by rhuss** [comment]
@iocanel okhttp 3.6.0 is out and it would be awesome if we could upgrade kubernetes-client. I tried my best, but we need probably some refactoring for the mock server as well as the API havs changed. 

People are eagerly waiting for better error reporting in f-m-p ;-) --> https://issues.jboss.org/browse/OSFUSE-557

File: kubernetes-client/pom.xml (hunk 1 above)
Comment (rhuss) [inline kubernetes-client/pom.xml]: I also think that okhttp-ws might not be needed anymore as it has been added to okhttp directly.

**Comment by iocanel** [comment]
@rhuss: sorry for the dealy. I completely missed that! Let me try to merge now.

**Comment by iocanel** [comment]
Merged your changes to the latest master as part of #702. So I am closing this one.

**Pull request closed without merging**